rtt/base/BufferLockFree.hpp

```diff
@@ -185,9 +185,15 @@ namespace RTT
                     // pop & deallocate until we have free space.
                     Item* itmp = 0;
                     do {
-                        bufs.dequeue( itmp );
-                        mpool.deallocate( itmp );
-                        droppedSamples.inc();
```

Comment:
Reverted `droppedSamples.inc()` in https://github.com/orocos-toolchain/rtt/commit/7fde6cbe677dbc46e58e8cdc2b9953da19fb0c2e: The new sample is not dropped, but the one removed from the circular buffer before. So incrementing the counter here is correct.